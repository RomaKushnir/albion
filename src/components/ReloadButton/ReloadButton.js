import { Button } from 'antd';
import { ReactComponent as ReloadIcon } from 'assets/images/reload-ico.svg';
import styles from './ReloadButton.module.scss';

const ReloadButton = ({ text, onUpdate }) => {
  return (
    <Button
      className={styles.btnWrap}
      type="primary"
      size="large"
      icon={<ReloadIcon />}
      onClick={() => onUpdate()}
    >
      {text}
    </Button>
  );
};

export default ReloadButton;
