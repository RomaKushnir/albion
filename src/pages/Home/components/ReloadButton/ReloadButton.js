import { Button } from 'antd';
import { ReactComponent as ReloadIcon } from 'assets/images/reload-ico.svg';
import styles from './ReloadButton.module.scss';

const ReloadButton = ({ onDataUpdate }) => {
  return (
    <Button
      className={styles.btnWrap}
      type="primary"
      size="large"
      icon={<ReloadIcon />}
      onClick={() => onDataUpdate()}
    >
      Update Leaderboard
    </Button>
  );
};

export default ReloadButton;
