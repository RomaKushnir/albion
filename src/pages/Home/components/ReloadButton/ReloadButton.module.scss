.btnWrap {
  display: flex;
  align-items: center;
  font-size: 18px;
  padding: 15px;
  border-radius: 0;
  svg {
    margin-right: 13px;
  }
}
