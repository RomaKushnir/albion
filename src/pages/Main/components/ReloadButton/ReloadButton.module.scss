.btnWrap {
  display: flex;
  align-items: center;
  font-size: 18px;
  svg {
    margin-right: 13px;
  }
}
