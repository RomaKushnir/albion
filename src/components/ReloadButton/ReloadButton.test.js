import ReloadButton from './ReloadButton';
import { shallow } from 'enzyme';

describe('render reload button', () => {
  const wrap = shallow(<ReloadButton />);

  wrap.setProps({ text: 'Reload', onUpdate: () => {} });

  it('should render correctly', () => {
    expect(wrap).toMatchSnapshot();
  });

  it('should reload button has the text', () => {
    expect(wrap.find('Button').text()).toEqual('Reload');
  });

  it('check is clicked', () => {
    wrap.find('Button').simulate('click');
    // should verify if the click is performed properly
    // expect()
  });
});
