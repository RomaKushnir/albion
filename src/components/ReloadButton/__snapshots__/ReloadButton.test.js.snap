// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render reload button should render correctly 1`] = `
<Button
  className="btnWrap"
  icon={<ForwardRef(SvgReloadIco) />}
  onClick={[Function]}
  size="large"
  type="primary"
>
  Reload
</Button>
`;
